packages/gatsby/src/commands/build.js

```diff
@@ -86,6 +86,20 @@ module.exports = async function build(program: BuildArgs) {
   })
   activity.end()
 
+  if (telemetry.isTrackingEnabled()) {
```

Comment:
This needs to be moved right under line 125 (`await processPageQueries()`) so that we know for sure that page queries are run (which would affect page data stats)